tests/test_runner/tests.py

```diff
@@ -404,3 +404,16 @@ def test_autoincrement_reset1(self):
     def test_autoincrement_reset2(self):
         p = Person.objects.create(first_name='Jack', last_name='Smith')
         self.assertEqual(p.pk, 1)
+
+
+class EmptyDefaultDatabaseTest(unittest.TestCase):
+    def test_empty_default_database(self):
+        """
+        Test that an empty default database in settings does not raise an ImproperlyConfigured
+        error when running a unit test that does not use a database.
+        """
+        db.connections = db.ConnectionHandler({'default': {}})
+        try:
+            connections_support_transactions()
+        except Exception as e:
+            self.fail("connections_support_transactions() unexpectedly raised an error: %s" % e)
```

Comment:
Seems like it would be useful to assert `db.connections[DEFAULT_DB_ALIAS]['ENGINE'] == 'django.db.backends.dummy'` per the logic you added in `db.utils`?